### ISSUE TITLE
Auto complete for commonly used words or words on the page already.

### ISSUE THREAD

**Issue by blankwall** [notable/notable#618]
Word auto complete similar to what is done in sublime. I would be willing to offer a donation on Patreon to get some priority on this and the spell checking feature. Thanks!

**Comment by fabiospampinato** [comment]
I'm not sure I would necessarily like to have this feature 🤔 Perhaps it could be implemented behind a setting flag #476. I'll keep this issue open to monitor the popularity of the request (number of "👍"s).

I do not prioritize issues based on donations since I'm trying to allocate my time towards the issues that will generally improve the user experience for most people. The issue about spell checking is well on the [radar](https://github.com/notable/notable/milestone/2) though.

**Comment by OnyxDelta** [comment]
I know this same feature exists in VSCode as well.  In that case, one can tab complete from a suggestion menu.  I think this would be very helpful.

**Comment by evansan** [comment]
Coding languages has much more string repetitions than natural language, I think this is not as useful. In order to keep Notable lean and not become bloated, we should not implement every feature,  only add essential features, while using plugins to implement less needed features.

**Comment by fabiospampinato** [comment]
I tend to agree with @evansan, but let's keep this issue open to track the popularity of the request.

Perhaps we could implement this only when editing inside a codeblocks, that could be pretty useful I think.

**Comment by blankwall** [comment]
As a use case example I write a lot of notes about code so I often find myself reusing the same words and syntax.

**Comment by fabiospampinato** [comment]
@blankwall once/if it's implemented for codeblocks, where IMHO this would mostly be useful for most people, we may very well add an option for enabling it in the entire note.
